stdlib/public/core/StringCore.swift

```diff
@@ -12,7 +12,7 @@
 
 /// The core implementation of a highly-optimizable String that
 /// can store both ASCII and UTF-16, and can wrap native Swift
-/// _StringBuffer or NSString instances.
+/// _StringBuffer or immutable literals.
```

Comment:
Not at the moment; it refers to Swift string literals stored in rodata, but the only assumption here is that the pointer lives "forever" and its contents are immutable. So if we can get such a pointer from objc, native Swift strings can wrap them.